DependencyInjection/Configuration.php

```diff
@@ -111,7 +111,14 @@ private function getMailersNode()
                     ->end()
                 ->end()
                 ->scalarNode('sender_address')->end()
-                ->scalarNode('delivery_address')->end()
+                ->arrayNode('delivery_addresses')
+                    ->beforeNormalization()
+                        ->always()
+                        ->then(function ($v) { return array_values((array) $v); })
```

Comment:
@xabbuh there is a long-standing bug in the Config component (but fixing it is a BC break, and I forogt it when we worked on 3.0, missing the opportunity for this): lists are reindexed numerically for prototyped nodes only when 2 configs need to be merged together. But if only 1 file is configuring it, we don't call ``array_values`` to reset keys.